### PR TITLE
Add logic to get active kuberetes pods from Kube API

### PR DESCRIPTION
**Pull request by thatInfrastructureGuy**
Added Tests and testdata.
Removed Log lines, returning error instead.
Refactored code to play well with tests.

### CODE REVIEW

**Review by JustinGaoF — DISMISSED** [review]
LGTM

**Review by gauthier-ca — DISMISSED** [review]

File: kubernetes/auth.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func (k *Config) authenticate() (err error) {
+	var config *rest.Config
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+	if len(kubeconfigPath) > 0 {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		log.Printf("Cannot connect to Kubernetes: %v\n", err)
```
Comment (taiyangc) [inline kubernetes/auth.go]: one small suggestion: either use `https://github.com/TRON-US/go-common/blob/master/db/postgres/hooks.go#L6` to do consistent logging with the rest of the modules, or return this message as an error directly and let upper level handle logging

Comment (thatInfrastructureGuy) [reply kubernetes/auth.go]: Done!

File: kubernetes/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	k.clientset = fake.NewSimpleClientset()
+	return &k
+}
+
+func TestAuthenticationDefaultError(t *testing.T) {
+	k := newFakeClient()
+	err := k.Authenticate()
+	if err == nil {
+		t.Error(err)
+	}
```
Comment (taiyangc) [inline kubernetes/auth_test.go]: Consider using testify, e.g. https://github.com/TRON-US/go-common/blob/master/operator/string_test.go#L27 it makes checks cleaner (forces you to write a reason on each failure), and also consistent with the rest of the tests.

Comment (thatInfrastructureGuy) [reply kubernetes/auth_test.go]: Done!

Comment (thatInfrastructureGuy) [reply kubernetes/auth_test.go]: Done!

**Review by taiyangc — APPROVED** [review]

**Pull request merged**